mmdet3d/datasets/scannet_dataset.py

```diff
@@ -277,3 +278,45 @@ def get_scene_idxs_and_label_weight(self, scene_idxs, label_weight):
 
         return super().get_scene_idxs_and_label_weight(scene_idxs,
                                                        label_weight)
+
+    def format_results(self, results, txtfile_prefix=None):
+        """Format the results to txt file. Refer to
+        http://kaldir.vc.in.tum.de/scannet_benchmark/documentation.
+
+        Args:
+            outputs (list[dict]): Testing results of the dataset.
+            txtfile_prefix (str | None): The prefix of saved files. It includes
+                the file path and the prefix of filename, e.g., "a/b/prefix".
+                If not specified, a temp file will be created. Default: None.
+
+        Returns:
+            tuple: (outputs, tmp_dir), outputs is the detection results,
+                tmp_dir is the temporal directory created for saving submission
+                files when ``submission_prefix`` is not specified.
+        """
+        import mmcv
+
+        if txtfile_prefix is None:
+            tmp_dir = tempfile.TemporaryDirectory()
+            txtfile_prefix = osp.join(tmp_dir.name, 'results')
+        else:
+            tmp_dir = None
+        mmcv.mkdir_or_exist(txtfile_prefix)
+
+        # need to map network output to original label idx
+        pred2label = np.zeros(len(self.VALID_CLASS_IDS)).astype(np.int)
+        for lbl, idx in self.label_map.items():
+            if idx != self.ignore_index:
+                pred2label[idx] = lbl
```

Comment:
try to use a more clear name rather than `lbl`.

Reply:
Fix.